src/components/figures/chart/chartJs/chart_js_extension.ts

```diff
@@ -1,12 +1,24 @@
-import { Plugin as ChartJSPlugin } from "chart.js";
 import { Registry } from "../../../../registries/registry";
 
-export const chartJsExtensionRegistry = new Registry<ChartJSPlugin>();
+export const chartJsExtensionRegistry = new Registry<{
+  register: (chart: typeof window.Chart) => void;
+  unregister: (chart: typeof window.Chart) => void;
+}>();
 
-/** Return window.Chart, making sure all our extensions are loaded in ChartJS */
-export function getChartJSConstructor() {
-  if (window.Chart && !window.Chart?.registry.plugins.get("chartShowValuesPlugin")) {
-    window.Chart.register(...chartJsExtensionRegistry.getAll());
+export function registerChartJSExtensions() {
+  if (!window.Chart || window.Chart.registry.plugins.get("chartShowValuesPlugin")) {
```

Comment:
I just have one issue with this strategy - we now HAVE to spawn a full spreadsheet to have the extensions available but we also use chartJs to export figures in XLSX  -  this means that we could theoretically have chartJs running alongside `Model` without any component. 

Could we imagine having this register return true if it had to actually do some work so we know that we need to unregister afterwards?  Then we could use it (in the future) for standalone models

Reply:
Rather than an opaque true/false return  value, I'd extract the `if` inside a `areChartJSExtensionLoaded()` I think. That would work for your use case no ?